qirlib/src/evaluation/runtime.rs

```diff
@@ -33,6 +41,15 @@ impl<'ctx> Simulator {
         gs.get_model()
     }
 
+    fn unsupported_function(module: &Module<'ctx>) -> Option<FunctionValue<'ctx>> {
+        module_functions(module).find(|f| {
+            let name = f.get_name().to_str().unwrap();
```

Comment:
Is there a safe way to avoid panicking during the `unwrap` here?

Reply:
Since this only fails if a function name contains non-utf8 chars (according to the docs https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#method.to_str), I assumed that we'd fail before reaching this point (probably while loading the module) and we're safe calling `unwrap` here.

If you're still thinking that this is a problem, we could catch and propagate the error.